tests/test_staticfiles.py

```diff
@@ -441,3 +441,26 @@ def mock_timeout(*args, **kwargs):
     response = client.get("/example.txt")
     assert response.status_code == 500
     assert response.text == "Internal Server Error"
+
+
+def test_staticfiles_follows_symlinks_to_break_out_of_dir(tmpdir, test_client_factory):
```

Comment:
what about we use the tmp_path fixture instead of tmpdir ? the former has a very similar api than the one pathlib provides and is imho more readable, opinions on that of course may vary so this is just a comment.